onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -2726,15 +2726,73 @@ def aten_index_copy(
     raise NotImplementedError()
 
 
+@torch_op("aten::index_put")
 def aten_index_put(
-    self: TensorType,
-    indices: Optional[Sequence[TensorType]],
-    values: TensorType,
+    self: TReal,
+    indices: Sequence[TReal],
+    values: TReal,
     accumulate: bool = False,
-) -> TensorType:
+) -> TReal:
     """index_put(Tensor self, Tensor?[] indices, Tensor values, bool accumulate=False) -> Tensor"""
 
-    raise NotImplementedError()
+    index = op.SequenceAt(indices, 0)  # assume indices only have 1 element
+    # change array([1,3]) to array([[1,1,1,1,1],[3,3,3,3,3]])
+    self_rank_1 = op.Gather(op.Shape(self), 1)
```

Comment:
For my education, what do rank_1 and rank_0 mean / do?

Reply:
rank_1 mean the size for dimension 1. rank_0 mean the size for dimension 0. 
shape(2,3),  rank_1 is 3, rank_0 is 2.
if you have a better name, pls share to me. :)

Reply:
Maybe call it dim? Rank to me can only be of a tensor but not a dimension